reproducible_build.sh

```diff
@@ -56,74 +80,258 @@ if [ "$CLONE" = false ]; then
     SOURCE_DIR=$PWD
 fi
 
-if [ -z ${TRAVIS+x} ]; then
-  # if not on travis, its nice to see progress
-  QUIET=""
+if [ -z ${TRAVIS_OS_NAME+x} ]; then
+    if [ `uname -s` = "Darwin" ]; then
+	OS_NAME="osx"
+    else
+	OS_NAME="linux"
+    fi
 else
-    QUIET="-qq"
+    OS_NAME=${TRAVIS_OS_NAME}
 fi
 
-# get the required OS packages
-$SUDO apt-get ${QUIET} update
-$SUDO apt-get ${QUIET} install -y --no-install-recommends \
-    build-essential python-dev libbz2-dev libtool \
-    autotools-dev autoconf git pkg-config wget \
-    ca-certificates automake bsdmainutils
-
-mkdir dependencies
-cd dependencies
-
-wget http://download.oracle.com/berkeley-db/db-4.8.30.NC.tar.gz
-tar xf db-4.8.30.NC.tar.gz
-export BDB_PREFIX="`pwd`/bdb"
-cd db-4.8.30.NC/build_unix
-../dist/configure --prefix=$BDB_PREFIX --enable-cxx --disable-shared --with-pic
-make
-make install
-cd ../../
-
-wget https://www.openssl.org/source/openssl-1.0.1p.tar.gz
-tar xf openssl-1.0.1p.tar.gz
-export OPENSSL_PREFIX="`pwd`/openssl_build"
-mkdir $OPENSSL_PREFIX
-cd openssl-1.0.1p
-./Configure --prefix=$OPENSSL_PREFIX --openssldir=$OPENSSL_PREFIX/ssl linux-x86_64 -fPIC -static no-shared no-dso
-make
-make install
-cd ..
-
-
-wget http://sourceforge.net/projects/boost/files/boost/1.59.0/boost_1_59_0.tar.bz2/download -O boost_1_59_0.tar.bz2
-tar xf boost_1_59_0.tar.bz2
-export BOOST_ROOT="`pwd`/boost_1_59_0"
-cd boost_1_59_0
-./bootstrap.sh
-./b2 link=static cxxflags=-fPIC stage
-cd ../../
-
-
-mkdir libevent_build
-git clone https://github.com/libevent/libevent.git
-export LIBEVENT_PREFIX="`pwd`/libevent_build"
-cd libevent
-./autogen.sh
-./configure --prefix=$LIBEVENT_PREFIX --enable-static --disable-shared --with-pic LDFLAGS="-L${OPENSSL_PREFIX}/lib/" CPPFLAGS="-I${OPENSSL_PREFIX}/include"
-make
-make install
-cd ..
+NEXT_TIME=60
+function exit_at_45() {
+    if [ -f ${HOME}/start_time ]; then
```

Comment:
I think it may be best to have a single cleanup() function that gets run when the script exits (via trap) and does all the cleanup. Then if you have to, you can modify that function's using variables.  

Or, you could do something like http://stackoverflow.com/questions/16115144/bash-save-and-restore-trap-state-easy-way-to-manage-multiple-handlers-for-trap